src/functions.c

```diff
@@ -186,18 +186,28 @@ void functionsLibCtxClearCurrent(int async) {
 }
 
 /* Free the given functions ctx */
-void functionsLibCtxFree(functionsLibCtx *functions_lib_ctx) {
+void functionsLibCtxFreeGeneric(functionsLibCtx *functions_lib_ctx, int async) {
+    if (async) {
+        freeFunctionsAsync(functions_lib_ctx);
+        return;
+    }
+
     functionsLibCtxClear(functions_lib_ctx);
     dictRelease(functions_lib_ctx->functions);
     dictRelease(functions_lib_ctx->libraries);
     dictRelease(functions_lib_ctx->engines_stats);
     zfree(functions_lib_ctx);
 }
 
+/* Free the given functions ctx */
+void functionsLibCtxFree(functionsLibCtx *functions_lib_ctx) {
+    functionsLibCtxFreeGeneric(functions_lib_ctx, 0);
+}
+
```

Comment:
There is still some circular call graph here, if we let functionsLibCtxFree call functionsLibCtxFreeGeneric. I suggest this:

```C
/* Free the given functions ctx */
void functionsLibCtxFreeGeneric(functionsLibCtx *functions_lib_ctx, int async) {
    if (async) {
        freeFunctionsAsync(functions_lib_ctx);
    } else {
        functionsLibCtxFree(functions_lib_ctx);
    }
}

/* Free the given functions ctx */
void functionsLibCtxFree(functionsLibCtx *functions_lib_ctx) {
    functionsLibCtxClear(functions_lib_ctx);
    dictRelease(functions_lib_ctx->functions);
    dictRelease(functions_lib_ctx->libraries);
    dictRelease(functions_lib_ctx->engines_stats);
    zfree(functions_lib_ctx);
}
```

Reply:
yes, thanks! it is better, i cant circular back my brain, it somehow stuck...

Reply:
I think you're too fast. :fast_forward: Easy to make mistake when you're very fast.